src/js/wrapper.ts

```diff
@@ -439,15 +456,31 @@ export const NATIVE: SentryNativeWrapper = {
    * @param key string
    * @param context key-value map
    */
-  setContext(key: string, context: { [key: string]: unknown } | null): void {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  setContext(key: string, context: { [key: string]: any } | null): void {
     if (!this.enableNative) {
       return;
     }
     if (!this._isModuleLoaded(RNSentry)) {
       throw this._NativeClientError;
     }
 
-    RNSentry.setContext(key, context !== null ? normalize(context) : null);
+    if (context === null) {
+      return RNSentry.setContext(key, null);
+    }
+
+    let normalizedContext: Record<string, unknown> | undefined;
+    try {
+      normalizedContext = convertToNormalizedObject(context);
+    } catch (e) {
+      logger.error('Context not passed to native SDK, because it contains non-serializable values', e);
```

Comment:
l: We could add the key to the error for easier debugging.
```suggestion
      logger.error('Context for key ${key} not passed to native SDK, because it contains non-serializable values', e);
```